airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/StandardSyncPersistence.java

```diff
@@ -214,4 +253,40 @@ private void writeStandardSync(final StandardSync standardSync, final DSLContext
     }
   }
 
+  private Stream<StandardSyncIdsWithProtocolVersions> findDisabledSyncs(final DSLContext ctx, final UUID actorDefId, final ActorType actorType) {
+    // Table aliasing to help have a readable join
+    final Actor source = ACTOR.as("source");
+    final Actor destination = ACTOR.as("destination");
+    final ActorDefinition sourceDef = ACTOR_DEFINITION.as("sourceDef");
+    final ActorDefinition destDef = ACTOR_DEFINITION.as("destDef");
+
+    // Retrieve all the connections currently disabled due to a bad protocol version
+    // where the actor definition is matching the one provided to this function
+    final Stream<StandardSyncIdsWithProtocolVersions> results = ctx
+        .select(CONNECTION.ID, sourceDef.ID, sourceDef.PROTOCOL_VERSION, destDef.ID, destDef.PROTOCOL_VERSION)
+        .from(CONNECTION)
+        .join(source).on(CONNECTION.SOURCE_ID.eq(source.ID))
+        .join(sourceDef).on(source.ACTOR_DEFINITION_ID.eq(sourceDef.ID))
+        .join(destination).on(CONNECTION.DESTINATION_ID.eq(destination.ID))
+        .join(destDef).on(destination.ACTOR_DEFINITION_ID.eq(destDef.ID))
+        .where(
+            CONNECTION.UNSUPPORTED_PROTOCOL_VERSION.eq(true).and(
+                (actorType == ActorType.DESTINATION ? destDef : sourceDef).ID.eq(actorDefId)))
+        .fetchStream()
+        .map(r -> new StandardSyncIdsWithProtocolVersions(
+            r.get(CONNECTION.ID),
+            r.get(sourceDef.ID),
+            AirbyteProtocolVersion.getWithDefault(r.get(sourceDef.PROTOCOL_VERSION)),
+            r.get(destDef.ID),
+            AirbyteProtocolVersion.getWithDefault(r.get(destDef.PROTOCOL_VERSION))));
+    return results;
+  }
+
+  private void clearProtocolVersionFlag(final DSLContext ctx, final List<UUID> standardSyncIds) {
+    ctx.update(CONNECTION)
```

Comment:
Nit: Please update the modified_at collumn as well. I don't think that is being updated through a trigger.

Reply:
Good catch, updated.